pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1280,6 +1323,46 @@ type ArtifactSearchQuery struct {
 	ArtifactName         string                      `json:"artifactName,omitempty" protobuf:"bytes,2,rep,name=artifactName"`
 	TemplateName         string                      `json:"templateName,omitempty" protobuf:"bytes,3,rep,name=templateName"`
 	NodeId               string                      `json:"nodeId,omitempty" protobuf:"bytes,4,rep,name=nodeId"`
+	Deleted              *bool                       `json:"deleted,omitempty" protobuf:"varint,5,opt,name=deleted"`
+}
+
+// map ArtifactGC Pod name to phase
+type ArtGCStatus struct {
+	// have Pods been started to perform this strategy?
+	StrategiesProcessed map[ArtifactGCStrategy]bool `json:"strategiesProcessed,omitempty" protobuf:"bytes,1,opt,name=strategiesProcessed"`
+
+	// have completed Pods been processed? (mapped by Pod name)
+	PodsRecouped map[string]bool `json:"podsRecouped,omitempty" protobuf:"bytes,2,opt,name=podsRecouped"`
```

Comment:
I don't understand this struct, maybe more comments and explanation would be good?